### ISSUE TITLE
ERROR  NEFF:neff_parse NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff

### ISSUE THREAD

**Issue by Sadden** [aws-neuron/aws-neuron-sdk#820]
Hi, 
I got following error when trying to compile llama2-13b-chat model

```
2024-Jan-18 01:49:40.972386 39717:39897 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972436 39717:39897 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.972467 39717:39888 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972464 39717:39887 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972441 39717:39901 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972568 39717:39901 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.9724352024-Jan-18 01:49:40.972528 39717:39881 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972595 39717:39900 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972694 39717:39900 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.972602 39717:39893 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972609 39717:39896 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
 39717:39888 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.972619 39717:39891 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.972587 39717:39897 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.972591 39717:39880 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.972606 39717:39889 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.9730982024-Jan-18 01:49:40.9731232024-Jan-18 01:49:40.972582 39717:39894 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.9725902024-Jan-18 01:49:40.972600 39717:39885 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.9725942024-Jan-18 01:49:40.9725992024-Jan-18 01:49:40.972840 39717:39893 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.972588 39717:39882 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
 39717:39889 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.972642 39717:39902 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff2024-Jan-18 01:49:40.973407 39717:39902 ERROR  NMGR:kmgr_load_nn_post_metrics               2024-Jan-18 01:49:40.972599 39717:39892 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
 39717:39899 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
 39717:39886 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.973485NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff 39717:39884 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.973537 39717:39884 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
 39717:39881 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 102024-Jan-18 01:49:40.972604 39717:39898 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.973437 39717:39895 ERROR  NEFF:neff_parse                               39717:39899 ERROR  NMGR:kmgr_load_nn_post_metrics               NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.973725 39717:39897 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff

2024-Jan-18 01:49:40.972609 39717:39883 ERROR  NEFF:neff_parse                              2024-Jan-18 01:49:40.973206 39717:39885 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.973454 39717:39892 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.973435 39717:39903 ERROR  NEFF:neff_parse                              NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff
2024-Jan-18 01:49:40.973961 39717:39903 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10

2024-Jan-18 01:49:40.9740392024-Jan-18 01:49:40.973063 39717:39880 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
NEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ffNEFF version: 2.0, features: 0x100 are not supported.  Currently supporting: 0x80000000000000ff

2024-Jan-18 01:49:40.973332
 39717:39882 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.973778Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10 39717:39891 ERROR  NMGR:kmgr_load_nn_post_metrics               2024-Jan-18 01:49:40.974129 39717:39898 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 102024-Jan-18 01:49:40.9726132024-Jan-18 01:49:40.9728972024-Jan-18 01:49:40.974163 39717:39894 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.973510 39717:39886 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
 39717:39895 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10

 39717:39887 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10


Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
 39717:39896 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
2024-Jan-18 01:49:40.974459 39717:39883 ERROR  NMGR:kmgr_load_nn_post_metrics               Failed to load NN: /tmp/no-user/neuroncc_compile_workdir/9fefc6db-2d43-4997-a88f-e44ef09a82b1/model.MODULE_f412a6bebf32d04bf191+2c2d707e.neff, err: 10
Traceback (most recent call last):
  File "/root/space/compile.py", line 32, in <module>
    model = NeuronModelForCausalLM.from_pretrained(huggingface_path, export=True, **compiler_args, **input_shapes)
  File "/opt/conda/lib/python3.10/site-packages/optimum/modeling_base.py", line 372, in from_pretrained
    return from_pretrained_method(
  File "/opt/conda/lib/python3.10/site-packages/optimum/neuron/modeling_decoder.py", line 155, in _from_transformers
    return cls._from_pretrained(checkpoint_dir, config)
  File "/opt/conda/lib/python3.10/site-packages/optimum/neuron/modeling_decoder.py", line 226, in _from_pretrained
    neuronx_model.to_neuron()
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/base.py", line 65, in to_neuron
    self.setup()
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/base.py", line 56, in setup
    nbs.setup()
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/decoder.py", line 332, in setup
    self.program.setup(self.layers, self.pre_layer_parameters, self.ln_lm_head_params)
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/decoder.py", line 1440, in setup
    super().setup(layers, pre_layer_params, ln_lm_head_params)
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/decoder.py", line 1318, in setup
    kernel.load(io_ring_cache_size)
  File "/opt/conda/lib/python3.10/site-packages/transformers_neuronx/compiler.py", line 444, in load
    self.model.load()
RuntimeError: nrt_load_collectives status=10


```

here is my code
```
huggingface_path = "meta-llama/Llama-2-13b-chat-hf"
    batch_size = 1
    num_cores = 24

    compiler_args = {"num_cores": num_cores, "auto_cast_type": 'fp16'}
    input_shapes = {"batch_size": batch_size, "sequence_length": 2048}
    model = NeuronModelForCausalLM.from_pretrained(huggingface_path, export=True, **compiler_args, **input_shapes)
```

Host: inf2-48xlarge
env: 
```
aws-neuronx-runtime-discovery==2.9
libneuronxla==0.5.669
neuronx-cc==2.12.54.0+f631c2365
neuronx-distributed==0.5.0
neuronx-hwm==2.12.0.0+422c9037c
optimum-neuron==0.0.16
torch-neuronx==1.13.1.1.13.0
torch-xla==1.13.1+torchneurond
transformers-neuronx==0.9.474
```

**Comment by aws-donkrets** [comment]
This error indicates that you have a compiler and runtime mismatch. the compiler is generating code that your currently installed runtime does not support. Suggest you update your runtime (or perhaps the entire install) to the latest SDK release.

**Comment by Sadden** [comment]
thanks. I tried to update or re-install all my neuron related dependencies but the issue persists. But once I tried to start a new docker container with the same image, it got auto-resolved. I think it could do something with the neuron cache

**Comment by hillarysanders** [comment]
@aws-donkrets I'm getting a similar error and am not really sure how to follow your advice. I went through the AWS documentation [here](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/setup/neuron-setup/pytorch/neuronx/ubuntu/torch-neuronx-ubuntu20-pytorch-dlami.html#setup-torch-neuronx-ubuntu20-dlami-pytorch) to update the neuron drivers, and the python package versions I'm using are the ones that the AMI (AWS's neuron ubuntu 20 PyTorch AMI (ami-055215a60bce6a9e5)).

--> What package versions are supposed to be used on these AWS supported AMIs, if not the ones that come pre-set? There's a lot of recommended packages flying around (optimum neuron, torch, torch-neuronx, transformers, transformers-neuronx, etc.). I tried upgrading torch-neuronx, but no nice. Is there a way to get a better error message here?
